lms/djangoapps/bulk_email/docs/decisions/001-bulk-email-content-sanitization.rst

```diff
@@ -17,16 +17,14 @@ It is considered good security practice to scan and sanitize user-provided conte
 Decision
 --------
 
-We will sanitize the HTML content received through the bulk course email tool before sending the messages to any recipients.
+We will not sanitize the HTML content received through the bulk course email tool on the back end. The content that instructors create for courses also allows unfiltered html, which is arguably a larger risk than email, which will block the execution of certain types of code. In the future, if a standard is set for filtering course content, the same standard could be applied here.
 
-We will use the `bleach`_ Python package to sanitize the data.
 
-We will introduce a new configuration setting called ``BULK_COURSE_EMAIL_ALLOWED_HTML_TAGS`` that acts as an *allowlist* of HTML tags permitted for use within the body of a message authored by the bulk course email tool. A default list of options is provided in the ``lms/envs/common.py`` `configuration file`_. Offending data will be escaped (converted to plaintext) over being stripped out.
+Rejected solutions
+------------------
 
-Consequences
-------------
+Bleach with allowlist ****************************
 
-A message sent through the Bulk Course Email tool that includes any restricted HTML content will not appear as intended to the recipients of the message. The restricted HTML content will be converted to plaintext and will not render.
+It has been standard practice for us to use Bleach with an allowlist to sanitize user provided content within the Open edX ecosystem. Santization using blocklists is vulnerable to obfuscation attacks, and the industry standard is to use an allowlist and explicitly enumerate all supported values. Given that this tool has been live for many years with no sanitization in place, a highly restrictive allowlist would be difficult to roll out to users, resulting in broken email templates and angry instructors who are used to having free rein. A permissive allowlist would potentially address this, but presents the non-trivial problem of assembling and maintaining a comprehensive list.
 
-.. _bleach: https://bleach.readthedocs.io/en/latest/
-.. _configuration file: https://github.com/openedx/edx-platform/blob/e608db847c39c2e3d723ef81f7dac66f63663a28/lms/envs/common.py#L4965
+_bleach: https://bleach.readthedocs.io/en/latest/
```

Comment:
hmm I bet that `..` is meaningful

Reply:
It is! We need to add the `..` back in. It's part of the hyperlink syntax so that `_bleach` bit won't appear at the bottom of the doc. Can see more here: https://github.com/ralsina/rst-cheatsheet/blob/master/rst-cheatsheet.rst.